libtock/lsm303dlhc.c

```diff
@@ -20,11 +20,22 @@ static void command_callback_yield (int data1, int data2, int data3, void* ud) {
 
 
 static int lsm303dlhc_subscribe (subscribe_cb cb, void *userdata) {
-  return subscribe(DRIVER_NUM_LSM303DLHC, 0, cb, userdata);
+  subscribe_return_t subval = subscribe2(DRIVER_NUM_LSM303DLHC, 0, cb, userdata);
+  if (subval.success == 0) {
+    return tock_error_to_rcode(subval.error);
+  }
+  return TOCK_SUCCESS;
 }
 
-static int lsm303dlhc_command (uint32_t command_num, uint32_t data1, uint32_t data2) {
-  return command(DRIVER_NUM_LSM303DLHC, command_num, data1, data2);
+static int lsm303dlhc_command_noval (uint32_t command_num, uint32_t data1, uint32_t data2) {
```

Comment:
What does "noval" mean? Shoudn't there be a "val" then?

Reply:
the lsm303dlhc driver never returns a `success_u32()`, so no need. noval in this case means "the kernel return a `CommandResult:success()` as opposed to any other success variant which contains a value"

Reply:
open to naming suggestions though

Reply:
A comment would be good. I found the specificity somewhat confusing, since there is no other command function in the file. As a reader I wonder, does "noval" matter? Why not specify other properties of the function in the name?

Reply:
done